pkg/utils/tekton/controller.go

```diff
@@ -96,31 +139,66 @@ func (s *SuiteController) ListTaskRuns(ns string, labelKey string, labelValue st
 	return s.PipelineClient().TektonV1beta1().TaskRuns(ns).List(context.TODO(), listOptions)
 }
 
-func (k KubeController) WatchTaskPod(tr string, taskTimeout int) error {
-	trUpdated, err := k.Tektonctrl.GetTaskRun(tr, k.Namespace)
-	if err != nil {
-		return err
-	}
-	pod, err := k.Commonctrl.GetPod(k.Namespace, trUpdated.Status.PodName)
-	if err != nil {
-		return err
-	}
-	return k.Commonctrl.WaitForPod(k.Commonctrl.IsPodSuccessful(pod.Name, k.Namespace), time.Duration(taskTimeout)*time.Second)
+func (k KubeController) WatchPipelineRun(pipelineRunName string, taskTimeout int) error {
+	return k.Commonctrl.WaitForPod(k.Tektonctrl.CheckPipelineRunFinished(pipelineRunName, k.Namespace), time.Duration(taskTimeout)*time.Second)
 }
 
-func (k KubeController) GetTaskRunResult(tr *v1beta1.TaskRun, result string) (string, error) {
-	for _, trResult := range tr.Status.TaskRunResults {
-		if trResult.Name == result {
-			return trResult.Value, nil
+func (k KubeController) GetTaskRunResult(pr *v1beta1.PipelineRun, result string) (string, error) {
+	for _, tr := range pr.Status.TaskRuns {
+		for _, trResult := range tr.Status.TaskRunResults {
+			if trResult.Name == result {
+				// for some reason the result might contain \n suffix
+				return strings.TrimSuffix(trResult.Value, "\n"), nil
```

Comment:
Rather than do this, we should probably just fix our tasks to not emit a newline character. Tekton saves the result as is.

Reply:
This can be done later on.